Sources/SwiftDocC/Infrastructure/Link Resolution/PathHierarchy+PathComponent.swift

```diff
@@ -170,56 +174,64 @@ private struct PathComponentScanner {
     }
     
     mutating func scanPathComponent() -> Substring {
-        // If the next component is an operator, parse the full operator before splitting on "/" ("/" may appear in the operator name)
-        if remaining.unicodeScalars.prefix(3).allSatisfy(\.isValidOperatorHead) {
-            guard let operatorEndIndex = remaining.firstIndex(of: Self.operatorEnd) else {
-                defer { remaining.removeAll() }
-                return remaining
-            }
-            
-            var component = remaining[..<operatorEndIndex]
-            remaining = remaining[operatorEndIndex...]
-            
-            guard let index = remaining.firstIndex(of: Self.separator) else {
-                defer { remaining.removeAll() }
-                return component + remaining
-            }
-            
-            component += remaining[..<index]
-            remaining = remaining[index...].dropFirst() // drop the slash
-            return component
-            
+        // If the next component is a Swift operator, parse the full operator before splitting on "/" ("/" may appear in the operator name)
+        if remaining.unicodeScalars.prefix(3).allSatisfy(\.isValidSwiftOperatorHead) {
+            return scanUntil(index: remaining.firstIndex(of: Self.swiftOperatorEnd))
+                 + scanUntilSeparatorAndThenSkipIt()
+        }
+        
+        // If the next component is a C++ operator, parse the full operator before splitting on "/" ("/" may appear in the operator name)
+        if remaining.starts(with: Self.cxxOperatorPrefix),
+           remaining.unicodeScalars.dropFirst(Self.cxxOperatorPrefixLength).first?.isValidCxxOperatorSymbol == true
+        {
+            return scan(length: Self.cxxOperatorPrefixLength)
+                 + scanUntil(index: remaining.unicodeScalars.firstIndex(where: { !$0.isValidCxxOperatorSymbol }))
```

Comment:
Note this does accept operator names that are non sensical e.g. `operator+-`.

Reply:
Yes, the path parser is a lot more permissive than the actual compilers. Its purpose is only to separate the authored link string into its components and each components into its name and disambiguation information. 

Conceptually you can think of it as a slightly more sophisticated version of `components(separatedBy: "/")` that knows not to split `operator/=` or `/=(_:_:)`.

---

In the case of a non sensical component like "operator+-", the path parser return that component to the path hierarchy (the core link resolution logic) which will try to find a matching node in the symbol hierarchy—relative to the starting point. When it can't find a matching node, it had more available information than the path parser had and can use this information to emit a diagnostic that "operator+-" wasn't found relative to the current page, with suggestions to replace it with "operator+" or "operator-" instead (assuming those operators exist in the symbol graph file). So, as a user-facing experience, returning the non sensical components here allows for more informative diagnostic messages in the end.